diff --git a/scripts/build/toolchain/termux_setup_toolchain_28c.sh b/scripts/build/toolchain/termux_setup_toolchain_28c.sh
index 829e429464..6b062a647e 100644
--- a/scripts/build/toolchain/termux_setup_toolchain_28c.sh
+++ b/scripts/build/toolchain/termux_setup_toolchain_28c.sh
@@ -1,7 +1,7 @@
 termux_setup_toolchain_28c() {
 	export CFLAGS=""
 	export CPPFLAGS=""
-	export LDFLAGS="-L${TERMUX__PREFIX__LIB_DIR}"
+	export LDFLAGS="-L${TERMUX__PREFIX__LIB_DIR} -Wl,-z,max-page-size=16384"
 
 	export AS=$TERMUX_HOST_PLATFORM-clang
 	export CC=$TERMUX_HOST_PLATFORM-clang
diff --git a/scripts/properties.sh b/scripts/properties.sh
index 82b1e6edca..5edf660fab 100644
--- a/scripts/properties.sh
+++ b/scripts/properties.sh
@@ -485,7 +485,7 @@ __termux_build_props__add_variables_validator_actions "TERMUX_APP__PACKAGE_NAME"
 #
 # Default value: `/data/data/com.termux`
 ##
-TERMUX_APP__DATA_DIR="/data/data/$TERMUX_APP__PACKAGE_NAME"
+TERMUX_APP__DATA_DIR="/data/data/com.liberivixer.youtubeharvester"
 __termux_build_props__add_variables_validator_actions "TERMUX_APP__DATA_DIR" "safe_absolute_path"
 
 ##
@@ -743,7 +743,7 @@ __termux_build_props__add_variables_validator_actions "TERMUX__ROOTFS_ID" "unsig
 #
 # Default value: `files`
 ##
-TERMUX__ROOTFS_SUBDIR="files"
+TERMUX__ROOTFS_SUBDIR=""
 __termux_build_props__add_variables_validator_actions "TERMUX__ROOTFS_SUBDIR" "allow_unset_value safe_relative_path"
 
@@ -784,5 +784,5 @@
 # Default value: `/data/data/com.termux/files`
 ##
-TERMUX__ROOTFS="$TERMUX_APP__DATA_DIR/$TERMUX__ROOTFS_SUBDIR"
+TERMUX__ROOTFS="$TERMUX_APP__DATA_DIR"
 TERMUX_BASE_DIR="$TERMUX__ROOTFS" # Deprecated alternative variable for `TERMUX__ROOTFS`

diff --git a/x11-packages/foot/build.sh b/x11-packages/foot/build.sh
--- a/x11-packages/foot/build.sh
+++ b/x11-packages/foot/build.sh
@@ -4,7 +4,7 @@
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="1.23.1"
 TERMUX_PKG_SRCURL=https://codeberg.org/dnkl/foot/archive/${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=02072b8f0aaf26907b6b02293c875539ce52fc59079344e7cf811ab03394cfa3
+TERMUX_PKG_SHA256=b3fa774983abb5f95aecca4557d146091d8666bc65fb310d4d3e38327357ade9
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="libandroid-support, fontconfig, freetype, libfcft, libpixman, libwayland, libxkbcommon, utf8proc"
 TERMUX_PKG_BUILD_DEPENDS="libtllist, libwayland-protocols, libwayland-cross-scanner, scdoc, xdg-utils"
diff --git a/packages/ncurses/build.sh b/packages/ncurses/build.sh
--- a/packages/ncurses/build.sh
+++ b/packages/ncurses/build.sh
@@ -17,3 +17,1 @@
-                    "$(. "$TERMUX_SCRIPTDIR/x11-packages/kitty/build.sh"; echo "$TERMUX_PKG_VERSION")"
-                    "$(. "$TERMUX_SCRIPTDIR/x11-packages/alacritty/build.sh"; echo "$TERMUX_PKG_VERSION")"
-                    "$(. "$TERMUX_SCRIPTDIR/x11-packages/foot/build.sh"; echo "$TERMUX_PKG_VERSION")")
+)
@@ -24,3 +21,1 @@
-                   "$(. "$TERMUX_SCRIPTDIR/x11-packages/kitty/build.sh"; echo "$TERMUX_PKG_SRCURL")"
-                   "$(. "$TERMUX_SCRIPTDIR/x11-packages/alacritty/build.sh"; echo "$TERMUX_PKG_SRCURL")"
-                   "$(. "$TERMUX_SCRIPTDIR/x11-packages/foot/build.sh"; echo "$TERMUX_PKG_SRCURL")")
+)
@@ -30,3 +23,1 @@
-                   "$(. "$TERMUX_SCRIPTDIR/x11-packages/kitty/build.sh"; echo "$TERMUX_PKG_SHA256")"
-                   "$(. "$TERMUX_SCRIPTDIR/x11-packages/alacritty/build.sh"; echo "$TERMUX_PKG_SHA256")"
-                   "$(. "$TERMUX_SCRIPTDIR/x11-packages/foot/build.sh"; echo "$TERMUX_PKG_SHA256")")
+)
@@ -114,7 +111,4 @@
-	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/a/{alacritty{,+common,-direct},ansi} "$TI/a/"
 	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/d/{dtterm,dumb} "$TI/d/"
 	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/e/eterm-color "$TI/e/"
-	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/f/foot{,+base,-direct} "$TI/f/"
 	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/g/gnome{,-256color} "$TI/g/"
 	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/n/nsterm "$TI/n/"
-	cp "$TERMUX_PKG_TMPDIR"/full-terminfo/k/kitty{,+common,-direct} "$TI/k/"
@@ -129,8 +122,2 @@
 	tic -x -o "$TI" "$TERMUX_PKG_SRCDIR/rxvt-unicode-${TERMUX_PKG_VERSION[1]}/doc/etc/rxvt-unicode.terminfo"
-	tic -x -o "$TI" "$TERMUX_PKG_SRCDIR/kitty-${TERMUX_PKG_VERSION[2]}/terminfo/kitty.terminfo"
-	tic -x -e alacritty,alacritty+common,alacritty-direct -o "$TI" "$TERMUX_PKG_SRCDIR/alacritty-${TERMUX_PKG_VERSION[3]}/extra/alacritty.info"
-
-	# Upstream instructions for building foot's terminfo
-	# See: https://codeberg.org/dnkl/foot/src/branch/master/INSTALL.md#terminfo
-	sed 's/@default_terminfo@/foot/g' "$TERMUX_PKG_SRCDIR/foot/foot.info" | \
-	tic -x -e foot,foot-direct -o "$TI" -
 }
diff --git a/packages/libx11/build.sh b/packages/libx11/build.sh
--- a/packages/libx11/build.sh
+++ b/packages/libx11/build.sh
@@ -11,1 +11,1 @@
-TERMUX_PKG_DEPENDS="libandroid-support, libxcb"
+TERMUX_PKG_DEPENDS="libandroid-shmem, libandroid-support, libxcb"
@@ -15,2 +15,3 @@
 ac_cv_path_RAWCPP=/usr/bin/cpp
+LIBS=-landroid-shmem
 --enable-malloc0returnsnull
diff --git a/packages/liblzo/build.sh b/packages/liblzo/build.sh
--- a/packages/liblzo/build.sh
+++ b/packages/liblzo/build.sh
@@ -7,2 +7,2 @@
-TERMUX_PKG_SRCURL=https://fossies.org/linux/misc/lzo-$TERMUX_PKG_VERSION.tar.xz
-TERMUX_PKG_SHA256=37ed4369e45944c53306b0d6a36b66f03e5b6aede8849c9b6388f4b62b20b443
+TERMUX_PKG_SRCURL=https://www.oberhumer.com/opensource/lzo/download/lzo-$TERMUX_PKG_VERSION.tar.gz
+TERMUX_PKG_SHA256=c0f892943208266f9b6543b3ae308fab6284c5c90e627931446fb49b4221a072
